src/V3Tristate.cpp

```diff
@@ -636,6 +639,129 @@ class TristateVisitor final : public TristateBaseVisitor {
+        }
+    }
+
+    uint8_t getStrength(AstAssignW* const nodep, bool value) {
+        if (AstStrengthSpec* const strengthSpec = nodep->strengthSpecp()) {
+            if (value)
+                return strengthSpec->strength1();
+            else
+                return strengthSpec->strength0();
+        } else {
+            return 6;  // default strength in strong (6)
+        }
+    }
+
+    bool assignmentOfValueOnAllBits(AstAssignW* const nodep, bool value) {
+        if (AstConst* const constp = VN_CAST(nodep->rhsp(), Const)) {
+            const V3Number num = constp->num();
+            if (value)
+                return num.isEqAllOnes();
+            else
+                return num.isEqZero();
+        }
+        return false;
+    }
+
+    AstAssignW* getStrongestAssignmentOfValue(const Assigns& assigns, bool value) {
+        auto maxIt = std::max_element(
+            assigns.begin(), assigns.end(), [&](AstAssignW* ap, AstAssignW* bp) {
+                bool valuesOnRhsA = assignmentOfValueOnAllBits(ap, value);
+                bool valuesOnRhsB = assignmentOfValueOnAllBits(bp, value);
+                if (!valuesOnRhsA && !valuesOnRhsB) return false;
+                if (!valuesOnRhsA) return true;
+                if (!valuesOnRhsB) return false;
+                return getStrength(ap, value) < getStrength(bp, value);
+            });
+        // If all assignment don't have const with all bits equal to value on the RHS,
+        // std::max_element will return one of them anyway, so we need to check before returning
+        if (assignmentOfValueOnAllBits(*maxIt, value))
+            return *maxIt;
+        else
+            return nullptr;
+    }
+
+    void removeWeakerAssignments(Assigns& assigns) {
+        // Weaker assignments are these assignments that can't change the final value of the net.
+        // If we know the value of the RHS, we take into account only strength corresponding to its
+        // value.
+
+        // First we want to find strongest assignment that has known value
+        AstAssignW* const strongest0p = getStrongestAssignmentOfValue(assigns, 0);
+        AstAssignW* const strongest1p = getStrongestAssignmentOfValue(assigns, 1);
+        AstAssignW* strongestp = nullptr;
+        uint8_t greatestKnownStrength;
+        if (strongest0p && strongest1p) {
+            strongestp = getStrength(strongest0p, 0) > getStrength(strongest1p, 1) ? strongest0p
+                                                                                   : strongest1p;
+
+            greatestKnownStrength
+                = std::max(getStrength(strongest0p, 0), getStrength(strongest1p, 1));
+        } else if (strongest0p) {
+            strongestp = strongest0p;
+            greatestKnownStrength = getStrength(strongest0p, 0);
+        } else if (strongest1p) {
+            strongestp = strongest1p;
+            greatestKnownStrength = getStrength(strongest1p, 1);
+        }
```

Comment:
```suggestion
        uint8_t greatestKnownStrength = 0;
        const auto getIfStrongest = [&](AstAssignW* const strongestCandidatep, bool value) {
            if (!strongestCandidatep) return;
            uint8_t strength = getStrength(strongestCandidatep, value);
            if (strength >= greatestKnownStrength) {
                greatestKnownStrength = strength;
                strongestp = strongest0p;
            }
        };
        getIfStrongest(strongest0p, 0);
        getIfStrongest(strongest1p, 1);
```

Reply:
Your way looks better, thanks. I see that I will only need to change `strongest0p` to `strongestCandidatep` in lambda's body.